packages/twenty-server/src/engine/core-modules/email-verification/email-verification.resolver.ts

```diff
@@ -23,15 +25,22 @@ export class EmailVerificationResolver {
     @OriginHeader() origin: string,
     @Context() context: I18nContext,
   ): Promise<ResendEmailVerificationTokenOutput> {
-    const workspace =
-      await this.domainManagerService.getWorkspaceByOriginOrDefaultWorkspace(
-        origin,
-      );
+    try {
+      const workspace =
+        await this.domainManagerService.getWorkspaceByOriginOrDefaultWorkspace(
+          origin,
+        );
 
-    return await this.emailVerificationService.resendEmailVerificationToken(
-      resendEmailVerificationTokenInput.email,
-      workspace,
-      context.req.headers['x-locale'] ?? SOURCE_LOCALE,
-    );
+      return await this.emailVerificationService.resendEmailVerificationToken(
+        resendEmailVerificationTokenInput.email,
+        workspace,
+        context.req.headers['x-locale'] ?? SOURCE_LOCALE,
+      );
+    } catch (error) {
+      if (error instanceof EmailVerificationException) {
```

Comment:
Not really related to this PR but I cam across EMAIL_NOT_VERIFIED and the ways it's handled seems off, it's at the root level while it should be an auth error probably?